Sources/PureSwiftJSON/Decoding/JSONUnkeyedDecodingContainer.swift

```diff
@@ -161,42 +157,32 @@ extension JSONUnkeyedDecodingContainer {
     }
 
     @inline(__always) private mutating func decodeFixedWidthInteger<T: FixedWidthInteger>() throws -> T {
-        defer {
-            currentIndex += 1
-            if currentIndex == count {
-                isAtEnd = true
-            }
-        }
-
-        guard case .number(let number) = self.array[self.currentIndex] else {
-            throw self.createTypeMismatchError(type: T.self, value: self.array[self.currentIndex])
+        let value = try self.getNextValue()
+        guard case .number(let number) = value else {
+            throw self.createTypeMismatchError(type: T.self, value: value)
         }
 
         guard let integer = T(number) else {
             throw DecodingError.dataCorruptedError(in: self,
                                                    debugDescription: "Parsed JSON number <\(number)> does not fit in \(T.self).")
         }
 
+        self.currentIndex += 1
         return integer
     }
 
-    @inline(__always) private mutating func decodeLosslessStringConvertible<T: LosslessStringConvertible>() throws -> T {
-        defer {
-            currentIndex += 1
-            if currentIndex == count {
-                isAtEnd = true
-            }
-        }
-
-        guard case .number(let number) = self.array[self.currentIndex] else {
-            throw self.createTypeMismatchError(type: T.self, value: self.array[self.currentIndex])
+    @inline(__always) private mutating func decodeBinaryFloatingPoint<T: LosslessStringConvertible>() throws -> T {
```

Comment:
I have been thinking about the naming here for quite some time... I'm not sure what the best way is forward...

Reply:
Honestly, I think it would be reasonable to just make it `<T: BinaryFloatingPoint>` instead of `<T: LosslessStringConvertible>`, since there's plenty of `LosslessStringConvertible` types the method won't work for anyway, so the existing conformance requirement is misleading.